# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpConnection.cpp
# Commit: f69657a7e597
# Full Hash: f69657a7e597e0cbf62f8dbcbebd8e7e230a9975
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-22 15:53:06
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Add a callback when the handshake is done. r=necko-reviewers,kershaw,keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123601
# ==============================================================================

diff -r e9635ec4198f -r f69657a7e597 netwerk/protocol/http/nsHttpConnection.cpp
--- a/netwerk/protocol/http/nsHttpConnection.cpp	Wed Sep 22 09:19:29 2021 +0000
+++ b/netwerk/protocol/http/nsHttpConnection.cpp	Wed Sep 22 09:19:30 2021 +0000
@@ -2368,6 +2368,7 @@
   NS_INTERFACE_MAP_ENTRY(nsIOutputStreamCallback)
   NS_INTERFACE_MAP_ENTRY(nsITransportEventSink)
   NS_INTERFACE_MAP_ENTRY(nsIInterfaceRequestor)
+  NS_INTERFACE_MAP_ENTRY(nsITlsHandshakeCallbackListener)
   NS_INTERFACE_MAP_ENTRY(HttpConnectionBase)
   NS_INTERFACE_MAP_ENTRY_CONCRETE(nsHttpConnection)
 NS_INTERFACE_MAP_END
@@ -2597,5 +2598,10 @@
   return val;
 }
 
+NS_IMETHODIMP
+nsHttpConnection::HandshakeDone() {
+  return NS_OK;
+}
+
 }  // namespace net
 }  // namespace mozilla